test/pieces.js

```diff
@@ -804,4 +815,124 @@ describe('Pieces', function() {
     }
   });
 
+  let token;
+  let bearerProductId;
+
+  it('should be able to log in as admin and get a bearer token', async () => {
+    // Log in
+    const response = await apos.http.post('/api/v1/@apostrophecms/login/login', {
+      body: {
+        username: 'admin',
+        password: 'admin'
+      }
+    });
+    assert(response.token);
+    token = response.token;
+  });
+
+  it('can POST a product with the bearer token', async () => {
```

Comment:
I tried to POST and PATCH a product having a join with an existing `article` id (with `"articlesIds": ["id-of-the-article"]`). No errors, but the join is not taken into account. Maybe the syntax has changed, but it would be useful to check that too.

Reply:
In A3, articlesIds is an implementation detail the developer should not think about. Instead you should actually pass `_articles`, as an array of objects. However the only mandatory properties of each object are `_id` and, if present, `_fields`. This is what allows us to implement the chooser/relationship field stuff more naturally in A3.

In A3 you patch schema fields, no matter how magical they may seem... not private implementation properties.